### PR TITLE
Gardenlet - fix Seed enqueue when Bootstrap status changed

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:

Fixes the Seed enqueue in the federated seed controller for Seeds that, when the Gardenlet starts, are not bootstrapped successfully. The Seed's condition looks like this:

```
 "conditions": [
            {
                "type": "Bootstrapped",
                "status": "False",
            }
```

The problem is that the federated seed controller's check if the Seed's status changed to True is flawed.

```
 "conditions": [
            {
                "type": "Bootstrapped",
                "status": "True",
            }
```
In fact that means that no federated seed controllers are started for the Seed --> no ShootStateSync or ControllerInstallation checks are being performed.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Fix the Seed enqueue in the federated seed controller for initially not-bootstrapped Seeds. Now Seeds are properly enqueued when their status changes from `not-bootstrapped` to `successfully bootstrapped`.
```

### CODE REVIEW

**Review by timebertt — APPROVED** [review]
/lgtm

**Comment by timebertt** [comment]
/kind/bug

**Comment by timebertt** [comment]
/topology/seed

File: pkg/gardenlet/controller/federatedseed/seed_control.go [hunk 1]
```diff
@@ -105,25 +105,16 @@ func (c *Controller) addSeedToQueue(obj interface{}) {
...
-	)
+func (c *Controller) seedUpdate(_, newObj interface{}) {
+	var newSeedObj = newObj.(*gardencorev1beta1.Seed)
 
 	newSeedBootstrappedCondition := helper.GetCondition(newSeedObj.Status.Conditions, gardencorev1beta1.SeedBootstrapped)
-	oldSeedBootstrappedCondition := helper.GetCondition(oldSeedObj.Status.Conditions, gardencorev1beta1.SeedBootstrapped)
-
-	if seedChangedToBootstrapped(newSeedBootstrappedCondition, oldSeedBootstrappedCondition) {
+	_, found := c.federatedSeedControllerManager.controllers[newSeedObj.Name]
+	if !found && newSeedBootstrappedCondition != nil && newSeedBootstrappedCondition.Status == gardencorev1beta1.ConditionTrue {
```
Comment (rfranzke) [inline pkg/gardenlet/controller/federatedseed/seed_control.go:113]: Hm, why didn't you just fix the bug in the old code? I think it makes sense to only enqueue on update if the condition switches to `True`, or?

Also, the check whether the controller is already started is now duplicated here (the reconciler is already checking it).

/cc @swilen-iwanow @plkokanov @vlvasilev

Comment (danielfoehrKn) [reply pkg/gardenlet/controller/federatedseed/seed_control.go:113]: I was thinking that what if we miss an update event? The controller would not be started any more. This way we could even miss the one particular update and would still start the controller at a later point in time. I just find it uncomfortable to rely only on one update event instead just checking if the desired state (having a controller) is fulfilled or not.

Yes the check is duplicated, but it is not an expensive one (lookup key in a map) vs [executing this function](https://github.com/gardener/gardener/blob/46a88c15db5397c573eba6f90f7cdba03327f361/pkg/gardenlet/controller/federatedseed/seed_control.go#L183-L182) and then performing the check.

Comment (rfranzke) [reply pkg/gardenlet/controller/federatedseed/seed_control.go:113]: OK, agreed. Makes sense, thanks!

**Review by rfranzke — APPROVED** [review]

**Pull request merged**